nx/lib/nx.ex

```diff
@@ -10906,6 +10906,43 @@ defmodule Nx do
     sqrt(variance(tensor, opts))
   end
 
+  @doc """
+  Remove all duplicate elements from 1D tensor.
+  ## Examples
+
+      iex> Nx.unique(Nx.tensor([1,1,2,3]))
+      #Nx.Tensor<
+        s64[3]
+        [1, 2, 3]
+      >
+
+     iex> Nx.unique(Nx.tensor([1,2,1,2,4,8]))
+     #Nx.Tensor<
+       s64[4]
+       [1, 2, 4, 8]
+     >
+
+     iex> Nx.unique(Nx.tensor([[1,2,3], [1,2,3]]))
+     ** (RuntimeError) expected a 1D tensor, got: 2D tensor
+  """
+  @doc type: :creation
+  @spec unique(tensor :: T.t()) :: T.t()
+  def unique(tensor) do
+    t = to_tensor(tensor)
+    t_rank = rank(tensor)
+
+    if t_rank == 1 do
+      t_items =
+        t
+        |> to_flat_list()
```

Comment:
Thanks for the proposal.

However, this function cannot currently be supported in Nx because the defn compiler must know the output shapes at compile-time, which is impossible for this function.

For future contributions, also keeo in mind that we try to stay on "Nx land" as much as possible because when dealing with GPUs and other devices, there's overhead in transferring data to the CPU, so this implementation would need to be changed as well.